From 733330888fff49e4d2b6c2121a6050fdd9f11a87 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Thu, 6 Feb 2020 09:32:04 +0800
Subject: [PATCH] rlm_python3: add PY_INC_DIR in search dir

The configure option --with-rlm-python3-include-dir is used to set
PY_INC_DIR which is never used and it fails to find Python.h,
so add it into search dir to fix it.

Also remove SMART_LIBS from mod_flags because it introduces rpath
to LDFALGS which causes a do_package_qa error:

ERROR: freeradius-3.0.20-r0 do_package_qa: QA Issue: package freeradius-python contains bad RPATH
/buildarea/build/tmp/work/core2-64-poky-linux/freeradius/3.0.20-r0/recipe-sysroot-native/usr/lib/python3.8/config in file
/buildarea/build/tmp/work/core2-64-poky-linux/freeradius/3.0.20-r0/packages-split/freeradius-python/usr/lib/rlm_python3.so.0.0.0
package freeradius-python contains bad RPATH
/buildarea/build/tmp/work/core2-64-poky-linux/freeradius/3.0.20-r0/recipe-sysroot-native/usr/lib/python3.8/config in file
/buildarea/build/tmp/work/core2-64-poky-linux/freeradius/3.0.20-r0/packages-split/freeradius-python/usr/lib/rlm_python3.so.0.0.0 [rpaths]

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 src/modules/rlm_python3/configure.ac | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/modules/rlm_python3/configure.ac b/src/modules/rlm_python3/configure.ac
index a00320f..adbdf19 100644
--- a/src/modules/rlm_python3/configure.ac
+++ b/src/modules/rlm_python3/configure.ac
@@ -95,7 +95,7 @@ if test x$with_[]modname != xno; then
 
 		old_CFLAGS=$CFLAGS
 		CFLAGS="$CFLAGS $PY_CFLAGS"
-		smart_try_dir="$PY_PREFIX/include/python$PY_SYS_VERSION"
+		smart_try_dir="$PY_PREFIX/include/python$PY_SYS_VERSION $PY_INC_DIR"
 		FR_SMART_CHECK_INCLUDE(Python.h)
 		CFLAGS=$old_CFLAGS
 
@@ -114,13 +114,13 @@ if test x$with_[]modname != xno; then
 
 		eval t=\${ac_cv_lib_${sm_lib_safe}_${sm_func_safe}}
 		if test "x$t" = "xyes"; then
-			mod_ldflags="$PY_LIB_LOC $PY_EXTRA_LIBS $SMART_LIBS -lm"
+			mod_ldflags="$PY_LIB_LOC $PY_EXTRA_LIBS -lm"
 			targetname=modname
 		else
 			FR_SMART_CHECK_LIB(python${PY_SYS_VERSION}m, Py_Initialize)
 			eval t=\${ac_cv_lib_${sm_lib_safe}_${sm_func_safe}}
 			if test "x$t" = "xyes"; then
-				mod_ldflags="$PY_LIB_LOC $PY_EXTRA_LIBS $SMART_LIBS -lm"
+				mod_ldflags="$PY_LIB_LOC $PY_EXTRA_LIBS -lm"
 				targetname=modname
 			else
 				targetname=
-- 
2.7.4

